fabric/job_queue.py

```diff
@@ -138,6 +140,11 @@ def _advance_the_queue():
 
         # Main loop!
         while not self._finished:
+            if self._abort:
```

Comment:
yeah, it might make sense to let the currently running jobs finish. If we want that behaviour, I don't even need to raise this exception. I can set queued to [] and let things end gracefully.